src/algebra/big_operators/finprod.lean

```diff
@@ -736,6 +756,24 @@ over `a ∈ ⋃₀ t` is the product over `s ∈ t` of the products of `f a` ove
   ∏ᶠ a ∈ ⋃₀ t, f a = ∏ᶠ s ∈ t, ∏ᶠ a ∈ s, f a :=
 by rw [set.sUnion_eq_bUnion, finprod_mem_bUnion h ht₀ ht₁]
 
+@[to_additive] lemma finprod_mul_sdiff_singleton (a : α) (hf : finite (mul_support f)) :
+  f a * (∏ᶠ i ∈ (mul_support f) \ {a}, f i) = ∏ᶠ i, f i  :=
```

Comment:
Is there any specific reason to use `i ∈ (mul_support f) \ {a}` instead of something like
```suggestion
  f a * (∏ᶠ i ≠ a, f i) = ∏ᶠ i, f i  :=
```

Reply:
I chose this formulation because when I apply this lemma in my project, I next have to rewrite  `∏ᶠ i ∈ (mul_support f) \ {a}, f i` to `∏ (i : a) in hf.to_finset \ {a}, f i`; if I start with `∏ᶠ i ≠ a, f i`, then to do this I have to use `finprod_cond_eq_prod_of_cond_iff`, and checking the hypotheses of this is a bit more annoying than just checking finiteness of `(mul_support f) \ {a}`. 
That being said, I agree that your formulation is cleaner. I could rewrite the statement to `f a * (∏ᶠ i ≠ a, f i) = ∏ᶠ i, f i`, and also add
`@[to_additive] lemma finprod_sdiff_singleton_eq_cond (a : α) (hf : finite (mul_support f)) :
  (∏ᶠ i ∈ (mul_support f) \ {a}, f i) = (∏ᶠ i ≠ a, f i)`; would that be a good solution?

Reply:
I think we should have a lemma to rewrite `∏ᶠ i ≠ a, f i` into `∏ i in hf.to_finset.erase a, f i`, which keeps `mul_support` out of the picture.

Reply:
Ah, that's probably better.